### PR TITLE
feat(VoiceState): add edit method

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds a new method: VoiceState#edit to allow both editable parameters to be edited in 1 API call. It also changes the existing parameters to return the current voicestate instead of void like before

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Comment by Jiralite** [comment]
Why wasn't this added to the `VoiceStateManager`?

File: packages/discord.js/src/structures/VoiceState.js [hunk 1]
```diff
@@ -206,28 +206,55 @@ class VoiceState extends Base {
...
+   * Currently only available to the logged in user's own voice state and when in a stage channel
+   * @param {VoiceStateEditData} data The data to edit the voice state with
+   * @returns {Promise<VoiceState>}
+   */
+  async edit(data) {
     if (this.channel?.type !== ChannelType.GuildStageVoice) throw new Error('VOICE_NOT_STAGE_CHANNEL');
 
     if (this.client.user.id !== this.id) throw new Error('VOICE_STATE_NOT_OWN');
 
+    if (typeof data.requestToSpeak !== 'boolean') throw new TypeError('VOICE_STATE_INVALID_TYPE', 'requestToSpeak');
```
Comment (muchnameless) [inline packages/discord.js/src/structures/VoiceState.js:226]: you should check if the properties exist, they are typed as optional

File: packages/discord.js/src/structures/VoiceState.js [hunk 2]
```diff
@@ -206,28 +206,55 @@ class VoiceState extends Base {
...
-   * // Making the client request to speak in a stage channel (raise its hand)
-   * guild.me.voice.setRequestToSpeak(true);
-   * @example
-   * // Making the client cancel a request to speak
-   * guild.me.voice.setRequestToSpeak(false);
-   * @returns {Promise<void>}
+   * Data to edit the logged in user's own voice state with, when in a stage channel
+   * @typedef {Object} VoiceStateEditData
+   * @property {boolean} [requestToSpeak] Whether or not the client is requesting to become a speaker.
+   * @property {boolean} [suppressed] Whether or not the user should be suppressed.
```
Comment (almeidx) [inline packages/discord.js/src/structures/VoiceState.js:212]:
```suggestion
   * @property {boolean} [requestToSpeak] Whether the client is requesting to become a speaker.
   * @property {boolean} [suppressed] Whether the user should be suppressed.
```

File: packages/discord.js/src/structures/VoiceState.js [hunk 3]
```diff
@@ -206,28 +206,55 @@ class VoiceState extends Base {
...
+        suppress: data.suppressed,
       },
     });
+    return this;
+  }
+
+  /**
+   * Toggles the request to speak in the channel.
+   * Only applicable for stage channels and for the client's own voice state.
+   * @param {boolean} [requestToSpeak=true] Whether or not the client is requesting to become a speaker.
```
Comment (almeidx) [inline packages/discord.js/src/structures/VoiceState.js:247]:
```suggestion
   * @param {boolean} [requestToSpeak=true] Whether the client is requesting to become a speaker.
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/VoiceState.js (hunk 2 above)
Comment (ImRodry) [reply packages/discord.js/src/structures/VoiceState.js:212]: I don't really feel like changing this. It's correct and I just copied the old docs, didn't edit them at all

File: packages/discord.js/src/structures/VoiceState.js [hunk 4]
```diff
@@ -240,26 +271,15 @@ class VoiceState extends Base {
    * // Making the client an audience member
    * guild.me.voice.setSuppressed(true);
    * @example
-   * // Inviting another user to speak
+   * // Inviting another user to speak\
```
Comment (vitorlops) [inline packages/discord.js/src/structures/VoiceState.js:274]: Why was this added?

Comment (ImRodry) [reply packages/discord.js/src/structures/VoiceState.js:274]: dunno

**Review by SpaceEEC — APPROVED** [review]

**Review by muchnameless — CHANGES_REQUESTED** [review]
~~I don't work with `VoiceState`s, but~~ the original methods had the following checks and Routes:
[setRequestToSpeak](https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/VoiceState.js#L223-L225)
`if (this.client.user.id !== this.id) throw new Error('VOICE_STATE_NOT_OWN');`
`Routes.guildVoiceState(this.guild.id)`
[setSuppressed](https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/VoiceState.js#L255-L257)
`const target = this.client.user.id === this.id ? '@me' : this.id;`
`Routes.guildVoiceState(this.guild.id, target)`
while the new [edit](https://github.com/ImRodry/discord.js/blob/feat/voicestate-edit/packages/discord.js/src/structures/VoiceState.js#L224-L234) method only has the first version, which to me seems like a regression

**Review by iCrawl — APPROVED** [review]

**Comment by ImRodry** [comment]
> ~I don't work with `VoiceState`s, but~ the original methods had the following checks and Routes:
> [setRequestToSpeak](https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/VoiceState.js#L223-L225)
> `if (this.client.user.id !== this.id) throw new Error('VOICE_STATE_NOT_OWN');`
> `Routes.guildVoiceState(this.guild.id)`
> [setSuppressed](https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/VoiceState.js#L255-L257)
> `const target = this.client.user.id === this.id ? '@me' : this.id;`
> `Routes.guildVoiceState(this.guild.id, target)`
> while the new [edit](https://github.com/ImRodry/discord.js/blob/feat/voicestate-edit/packages/discord.js/src/structures/VoiceState.js#L224-L234) method only has the first version, which to me seems like a regression

You’re right, I’ll fix this later today

**Comment by ImRodry** [comment]
Fixed and updated documentation to state that only requestToSpeak is exclusive to the logged in client

File: packages/discord.js/src/structures/VoiceState.js [hunk 5]
```diff
@@ -205,29 +205,64 @@ class VoiceState extends Base {
...
+   * Edits this voice state. Currently only available when in a stage channel
+   * @param {VoiceStateEditData} data The data to edit the voice state with
+   * @returns {Promise<VoiceState>}
+   */
+  async edit(data) {
+    if (this.channel?.type !== ChannelType.GuildStageVoice) throw new Error('VOICE_NOT_STAGE_CHANNEL');
+
+    const target = this.client.user.id === this.id ? '@me' : this.id;
+
+    if (target === '@me' && typeof data.requestToSpeak !== 'undefined') {
```
Comment (muchnameless) [inline packages/discord.js/src/structures/VoiceState.js:226]:
```suggestion
    if (target !== '@me' && typeof data.requestToSpeak !== 'undefined') {
```

**Review by muchnameless — CHANGES_REQUESTED** [review]
